### DISCUSSION TITLE
Get cell metada from notebook cell renderer

### DISCUSSION THREAD

**Discussion by HatemMn** [microsoft/vscode-discussions#817] [Extension Development QnA] [answered]
As indicated in the title, my goal is straigthforward : 
I have a notebook renderer, that renders the output of some cell in a specefic manner using React.
The renderer has access to the output of the cell, but how can he get the metadata of that same cell ?

```
export const activate: ActivationFunction = function (
  context: RendererContext<unknown>,
) {
  return {

    renderOutputItem(data, element, _signal) {
const cellOutput = data.json(); // but how to get the metadata ?
}

}
```

PS : I tried to include vscode API and call it, but could not

**Comment by theReynald** [comment, marked as answer]
Hi @hatemMn,

In a VS Code extension, when working with a notebook renderer, you can access the metadata of a cell from the `data` object provided to the `renderOutputItem` function. However, the exact structure of the `data` object depends on the specific renderer and how the data is passed.

Typically, in a notebook renderer, you can access the metadata of the cell using `data.cell.metadata`. Here's how you can modify your code to access the cell metadata:

```typescript
export const activate: ActivationFunction = function (context: RendererContext<unknown>) {
    return {
        renderOutputItem(data, element, _signal) {
            // Access the metadata of the cell
            const cellMetadata = data.cell.metadata;

            // Now you can work with cellMetadata as needed
        }
    };
};
```

In the code above, `data.cell.metadata` should provide you with access to the metadata of the cell being rendered. This assumes that the renderer framework you are using follows the standard pattern for passing cell data to the renderer. If your specific renderer behaves differently, you may need to adapt the code accordingly.

Keep in mind that the structure and availability of cell metadata can vary depending on the notebook format and the specific extension or renderer you're working with. Be sure to check out the documentation or source code of the renderer you're using to understand how cell data and metadata are provided.

Hope this helped.
